lib/core/src/system/web/text_input.rs

```diff
@@ -0,0 +1,130 @@
+//! This module contains Rust wrappers for keyboard event handling.
+//!
+//! These keyboard events are taken from created invisible textarea element in html document body.
+//! We do it this way, because this is the only way for handling clipboard operations.
+
+use std::fmt::Debug;
+use std::fmt::Formatter;
+use std::fmt::Error;
+use wasm_bindgen::prelude::*;
+use web_sys::KeyboardEvent;
+
+
+
+// ===========================
+// === JavaScript Bindings ===
+// ===========================
+
+/// A module with functions imported from JavaScript.
+mod js {
+    use wasm_bindgen::prelude::*;
+    use web_sys::KeyboardEvent;
+
+    #[wasm_bindgen(module = "/src/system/web/text_input/text_input.js")]
+    extern "C" {
+        pub type TextInputHandlers;
+
+        #[allow(unsafe_code)]
+        #[wasm_bindgen(constructor)]
+        pub fn new() -> TextInputHandlers;
+
+        #[allow(unsafe_code)]
+        #[wasm_bindgen(method)]
+        pub fn set_copy_handler(this:&TextInputHandlers, handler:&Closure<dyn FnMut(bool) -> String>);
+
+        #[allow(unsafe_code)]
+        #[wasm_bindgen(method)]
+        pub fn set_paste_handler(this:&TextInputHandlers, handler:&Closure<dyn FnMut(String)>);
+
+        #[allow(unsafe_code)]
+        #[wasm_bindgen(method)]
+        pub fn set_event_handler
+        (this:&TextInputHandlers, name:&str, handler:&Closure<dyn FnMut(KeyboardEvent)>);
+
+        #[allow(unsafe_code)]
+        #[wasm_bindgen(method)]
+        pub fn stop_handling(this:&TextInputHandlers);
+    }
+}
+
+
+
+// =======================
+// === KeyboardBinding ===
+// =======================
+
+/// Copy handler takes bool which is true on cut operations, and returns the string should be
+/// actually copied to clipboard.
+pub trait CopyHandler          = FnMut(bool) -> String + 'static;
+
+/// The paste handler takes in its argument the text from clipboard to paste.
+pub trait PasteHandler         = FnMut(String) + 'static;
+
+/// Keyboard event handler takes event as an argument.
+pub trait KeyboardEventHandler = FnMut(KeyboardEvent) + 'static;
```

Comment:
Lets not use alignment in case these are not glued together. It looks visually bad imo.

Reply:
I align first, then add docs later, that's why I'm so often leaving it this way.

Reply:
Yeah, I totally get it. I do the same thing. :)